src/docfx/build/xref/XrefMap.cs

```diff
@@ -152,24 +156,28 @@ private bool TryResolveFromExternal(string uid, out XrefSpec spec)
 
         public static XrefMap Create(Context context, Docset docset)
         {
-            Dictionary<string, XrefSpec> map = new Dictionary<string, XrefSpec>();
-            foreach (var url in docset.Config.Xref)
+            // TODO: not considering same uid with multiple specs, it will be Dictionary<string, List<T>>
+            // https://github.com/dotnet/corefx/issues/12067
```

Comment:
👍 want to replace all `ConcurrentBags`, `ConcurrentDictionary` with non concurrent ones and locks?

Reply:
That depends on the scenario while reading and adding, `ConcurrentDictionary` is better. But sure, we can take a look at the other usages